pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -300,7 +300,7 @@ func translateInTreeSpecToCSIIfNeeded(spec *volume.Spec, nodeName types.NodeName
 		return nil, err
 	}
 	if migratable && migrationSupportedOnNode {
-		translatedSpec, err = csimigration.TranslateInTreeSpecToCSI(spec, csiTranslator)
+		translatedSpec, err = csimigration.TranslateInTreeSpecToCSI(spec, podNamespace, csiTranslator)
```

Comment:
Note that for attach/detach, API changes to VolumeAttachment are needed if you need a secret for attach/detach. In this case, the secret is only needed for NodePublish right?

Reply:
correct, actually it's only for `NodeStageSecretRef`